opfpush/src/main/java/org/onepf/opfpush/listener/EventListener.java

```diff
@@ -37,48 +37,61 @@
+     * {@link org.onepf.opfpush.OPFConstants#MESSAGES_COUNT_UNKNOWN}
      *
      * @param context       application context.
-     * @param providerName  Name of provider received event.
-     * @param messagesCount Count of messages. Negative value if no info about count.
+     * @param providerName  Name of provider which has received event.
+     * @param messagesCount Count of messages. {@link org.onepf.opfpush.OPFConstants#MESSAGES_COUNT_UNKNOWN} value
+     *                      if there isn't information about count.
      */
     void onDeletedMessages(@NonNull Context context, @NonNull String providerName, int messagesCount);
 
     /**
-     * Provider registered successfully.
+     * Called when message about successful registration has been received.
      *
      * @param context        application context.
-     * @param providerName   Name of registered provider.
-     * @param registrationId Registration id for push notification.
+     * @param providerName   Name of provider which has received event.
+     * @param registrationId Registration ID for push notifications.
      */
     void onRegistered(@NonNull Context context, @NonNull String providerName, @NonNull String registrationId);
 
     /**
-     * Provider unregistered successfully.
+     * Called when {@link org.onepf.opfpush.OPFPushHelper} become to {@code UNREGISTERED} state.
+     * Calling of this method doesn't mean that provider is already unregistered. Unregistration
+     * is performed in the background and is retried if unregistration error is occurred.
      *
      * @param context        application context.
-     * @param providerName   Name of unregistered provider.
-     * @param registrationId Old registration id for push notification.
+     * @param providerName   Name of provider which has received event.
+     * @param registrationId Old registration ID for push notification.
      */
     void onUnregistered(@NonNull Context context, @NonNull String providerName, @Nullable String registrationId);
 
     /**
      * {@code OpenPushHelper} can't find any available provider for register push.
-     * Notify user that push notifications will not be received.
-     * Try to add more push providers to opfpush configurations.
+     * You should notify user that push notifications will not be received.
+     * <p/>
+     * A push provider can be unavailable in two common reasons:
+     * 1) {@link org.onepf.opfpush.PushProvider#isAvailable()} method has returned false value.
+     * 2) Some unrecoverable registration error has been occurred.
+     * In the second case you can get {@link org.onepf.opfpush.model.OPFError} object from {@code registrationErrors}
+     * map. You can notify user about occurred error, if the error can be resolved by user.
+     * For example if you get {@link org.onepf.opfpush.model.OPFError#AUTHENTICATION_FAILED} for GCM push provider,
+     * you can ask user add the google account.
```

Comment:
you can ask user whether he wants to add the google account.
you can ask user to add the google account.